src/platform/windows/monitor.rs

```diff
@@ -52,39 +56,265 @@ fn wchar_as_string(wchar: &[wchar_t]) -> String {
 
 unsafe extern "system" fn monitor_enum_proc(hmonitor: HMONITOR, _: HDC, place: LPRECT, data: LPARAM) -> BOOL {
     let monitors = data as *mut VecDeque<MonitorId>;
-
     let place = *place;
     let position = (place.left as i32, place.top as i32);
     let dimensions = ((place.right - place.left) as u32, (place.bottom - place.top) as u32);
-
     let mut monitor_info: winuser::MONITORINFOEXW = mem::zeroed();
     monitor_info.cbSize = mem::size_of::<winuser::MONITORINFOEXW>() as DWORD;
+
     if winuser::GetMonitorInfoW(hmonitor, &mut monitor_info as *mut winuser::MONITORINFOEXW as *mut winuser::MONITORINFO) == 0 {
         // Some error occurred, just skip this monitor and go on.
         return TRUE;
     }
 
+    let monitor_name = wchar_as_string(&monitor_info.szDevice); // "\\\\.\\DISPLAY1"
+    let physical_size = get_physical_size_for_monitor(&monitor_name).and_then(|(w, h)| Some((w as u64, h as u64)));
+
     (*monitors).push_back(MonitorId {
         adapter_name: monitor_info.szDevice,
         hmonitor: HMonitor(hmonitor),
-        monitor_name: wchar_as_string(&monitor_info.szDevice),
+        monitor_name,
         primary: monitor_info.dwFlags & winuser::MONITORINFOF_PRIMARY != 0,
         position,
         dimensions,
         hidpi_factor: 1.0,
+        extents_mm: physical_size,
     });
 
     // TRUE means continue enumeration.
     TRUE
 }
 
+/// Get the physical size of a monitor from the Windows registry
+///
+/// The `monitor_name` is obtained from the `MONITORINFOEXQ::szDevice` field.
+fn get_physical_size_for_monitor(monitor_name: &str) -> Option<(u16, u16)> {
+
+    use winapi::shared::minwindef::HKEY;
+    use winapi::shared::winerror::ERROR_NO_MORE_ITEMS;
+    use winapi::shared::guiddef::GUID;
+    use winapi::um::setupapi::{
+        SetupDiGetClassDevsExW, SetupDiDestroyDeviceInfoList, SetupDiEnumDeviceInfo,
+        SetupDiGetDeviceInstanceIdW, SetupDiOpenDevRegKey, DICS_FLAG_GLOBAL, DIREG_DEV};
+    use winapi::um::errhandlingapi::GetLastError;
+    use winapi::um::handleapi::INVALID_HANDLE_VALUE;
+    use winapi::um::setupapi::{DIGCF_PRESENT, DIGCF_PROFILE, SP_DEVINFO_DATA};
+    use winapi::um::winnt::KEY_READ;
+    use winapi::um::winreg::RegCloseKey;
+    use winapi::um::cfgmgr32::MAX_DEVICE_ID_LEN;
+
+    // Find the display_device with the correct monitor name
+    let display_device = get_display_device_from_monitor_name(monitor_name)?;
+    let device_id = wchar_as_string(&display_device.DeviceID[..]);
+    println!("device id - {:?}, monitor name: {:?}", device_id, monitor_name);
```

Comment:
Should be removed or replaced by logging.

Reply:
Thanks for helping out, but this PR is WIP, so I don't think this is pertinent.

(Also, winit has no logging infrastructure whatsoever, so your proposed alternative would be something of an undertaking. If you want winit to have logging sooner, then I recommend opening an issue so everyone has a chance to argue about whether or not to use `slog`. It's something I've been meaning to do, but I've been busy.)